laituri/docker/credential_manager/ecr_with_role_v1.py

```diff
@@ -0,0 +1,49 @@
+import base64
+from contextlib import contextmanager
+from typing import Iterator
+
+import requests
+
+from laituri.docker.credential_manager.docker_v1 import docker_v1_credential_manager
+from laituri.docker.credential_manager.errors import CallbackFailed, DockerLoginFailed
+from laituri.types import LogStatusCallable, RegistryCredentialsDict
+
+
+@contextmanager
+def ecr_with_role_v1_credential_manager(
+    *,
+    image: str,
+    registry_credentials: RegistryCredentialsDict,
+    log_status: LogStatusCallable,
+) -> Iterator[None]:
+    from boto3 import Session
+    aws_imds_v2_token = requests.request(
+        'PUT',
+        'http://169.254.169.254/latest/api/token',
+        headers={'X-aws-ec2-metadata-token-ttl-seconds': '21600'},
+        timeout=1
+    ).text
```

Comment:
This will _force_ using IMDSv2; should only do it the request fails without the token.

Could we use `botocore.utils.InstanceMetadataFetcher` here?

Reply:
More generally, cribbing from `botocore.credentials`:

```
fetcher = InstanceMetadataFetcher(
    #timeout=metadata_timeout,
    #num_attempts=num_attempts,
    #user_agent=session.user_agent(),
    #config=imds_config,
)

# check role stuff?

instance_metadata_provider = InstanceMetadataProvider(iam_role_fetcher=fetcher)
credentials = instance_metadata_provider.load()
sess = Session(..., credentials=credentials)
```